react-app/src/components/Extractor.js

```diff
@@ -0,0 +1,303 @@
+                  extraButton={
+                    <Button
+                      onClick={() => {
+                        const newArgs = [...args];
+                        newArgs.find((temp) => temp.key === 'filePath').included = false;
+                        newArgs.find((temp) => temp.key === 'url').included = true;
+                        updateArgData(newArgs, false);
+                      }}
+                      className="generic-button narrow-button"
+                      variant="outline-info"
+                    >
+                      Switch to URL
+                    </Button>
+                  }
+                />
+              </div>
+            );
+          case 'url':
+            return (
+              <Form.Group className="mb-3" controlId={arg.key} key={arg.key}>
+                <Form.Label>{arg.label}</Form.Label>
+                <div className="file-button-container">
+                  <Button
+                    onClick={() => {
+                      const newArgs = [...args];
+                      newArgs.find((temp) => temp.key === 'url').included = false;
+                      newArgs.find((temp) => temp.key === 'filePath').included = true;
+                      updateArgData(newArgs, false);
+                    }}
+                    className="generic-button narrow-button"
+                    variant="outline-info"
+                  >
+                    Switch to CSV File
+                  </Button>
+                  <Form.Control
+                    type="text"
+                    value={arg[arg.key]}
+                    onChange={(e) => {
+                      const newArgs = [...tempArgs];
+                      newArgs.find((temp) => arg.key === temp.key)[arg.key] = e.target.value;
+                      updateArgData(newArgs, false);
+                    }}
+                    placeholder="Enter URL"
+                  />
+                </div>
+              </Form.Group>
+            );
+          case 'dropdown':
+            return (
+              <Form.Group className="mb-3" controlId={arg.key} key={arg.key}>
+                <div>
+                  <Form.Label>{arg.label}</Form.Label>
+                </div>
+                <Form.Text>
+                  Enter field(s) as a list separated by a commas. The options are gender, mrn, name, address, birthDate,
+                  language, ethnicity, birthsex, and race.
+                </Form.Text>
+                <div className="label-and-icon-container">
+                  <Form.Control
+                    type="text"
+                    value={arg[arg.key]}
+                    onChange={(e) => {
+                      const newArgs = [...tempArgs];
+                      newArgs.find((temp) => arg.key === temp.key)[arg.key] = e.target.value
+                        .split(',')
+                        .map((word) => word.trim());
+                      updateArgData(newArgs, false);
```

Comment:
If possible, I think it would be good to separate the manipulation you do on the string from what is actually displayed in the UI. With this how it is now, the input field swallows the spaces when you type them, which is a little jarring. Also, if you try to hit enter while typing in the middle of the string, the cursor is moved to the end, which isn't expected. I think it would be good if we only did the `split` and `trim` when we use the formData to actually output the configuration file, and not update what is actually input into the box.

I'm not sure if we actually want to or can fix this now, but it might be good to be aware of when we hook up with submit button and access all the data from the form.

Reply:
I can leave it as a regular string input for now. Then I can add the code to convert the string -> array while working on STEAM-723.
I have removed the calls to split and trim, so now it just saves the string.